core/options.c

```diff
@@ -2379,8 +2379,8 @@ options_init()
     /* .lspdata pages start out writable so no unprotect needed here */
     write_lock(&options_lock);
     ASSERT(sizeof(dynamo_options) == sizeof(options_t));
-    /* get dynamo options */
-    adjust_defaults_for_page_size(&dynamo_options);
+    /* reset options to the defaults and get current dynamo options */
+    set_dynamo_options_defaults(&dynamo_options);
```

Comment:
This is a big memcpy.  I'd prefer to do this on exit and only if detaching, like we're doing the other expensive zeroing.

Reply:
There's a comment about how options_exit doesn't clear option values because we need them later: https://github.com/DynamoRIO/dynamorio/blob/de99d4540dc3196fb4635fbae1cbc1a685aa1e10/core/dynamo.c#L1114

Should we have another function "options_cleanup" which is only called in detach code, and only at the very end? Seems like it'd be easy to mess up later.

Reply:
Hmm, ok, or `options_detach`